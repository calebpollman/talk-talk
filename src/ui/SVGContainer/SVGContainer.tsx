import { ReactNode, SVGAttributes } from 'react';

interface SVGContainerProps extends Omit<SVGAttributes<SVGElement>, 'children' | 'height' | 'width'> {
  children: ReactNode;
}

const CONTAINER_DIMENSION = '96%';

function SVGContainer({ children, ...rest }: SVGContainerProps) {
  return (
    <svg {...rest} width={CONTAINER_DIMENSION} height={CONTAINER_DIMENSION}>
      {children}
    </svg>
  );
}

export default SVGContainer;
