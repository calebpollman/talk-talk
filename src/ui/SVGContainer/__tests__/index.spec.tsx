import { render } from '@testing-library/react';

import SVGContainer from '..';

function TestChildren() {
  return null;
}

describe('SVGContainer', () => {
  it('renders as expected', () => {
    const { baseElement } = render(
      <SVGContainer>
        <TestChildren />
      </SVGContainer>
    );

    expect(baseElement).toMatchSnapshot();
  });
});
