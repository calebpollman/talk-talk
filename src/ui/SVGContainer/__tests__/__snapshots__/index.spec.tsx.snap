// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVGContainer renders as expected 1`] = `
<body>
  <div>
    <svg
      height="96%"
      width="96%"
    />
  </div>
</body>
`;
